src/java.desktop/windows/classes/sun/awt/windows/ThemeReader.java

```diff
@@ -111,11 +121,16 @@ private static Long getTheme(String widget) {
             try {
                 if (!valid) {
                     // Close old themes.
-                    for (Long value : widgetToTheme.values()) {
-                        closeTheme(value);
+                    if (!dpiAwareWidgetToTheme.isEmpty()) {
+                        for (Map<String, Long> dpiVal : dpiAwareWidgetToTheme.values()) {
+                            for (Long value : dpiVal.values()) {
+                                closeTheme(value);
+                            }
+                        }
+                        dpiAwareWidgetToTheme.get(dpi).clear();
+                        dpiAwareWidgetToTheme.clear();
```

Comment:
```suggestion
                        for (Map<String, Long> dpiVal : dpiAwareWidgetToTheme.values()) {
                            for (Long value : dpiVal.values()) {
                                closeTheme(value);
                            }
                            dpiVal.clear();
                        }
                        dpiAwareWidgetToTheme.clear();
```
Avoid additional call to `get`.

Reply:
This is even more serious: all the theme handles are closed but the maps for DPIs other than the passed `dpi` aren't cleared. With `clear` inside the loop, maps for all DPIs are cleared.

Reply:
I've thought about it more, and I was **wrong**.

Clearing the `dpiVal` map isn't needed at all because as the next step the outer map gets cleared, meaning that maps with theme handles are thrown away. Thus, no need to waste time to run `dpiVal.clear()` — `dpiAwareWidgetToTheme.clear()` takes care of those too.

That is remove the line `dpiVal.clear()`, please.